src/couch/src/couch_bt_engine.erl

```diff
@@ -192,6 +193,44 @@ handle_db_updater_info({'DOWN', Ref, _, _, _}, #st{fd_monitor=Ref} = St) ->
     {stop, normal, St#st{fd=undefined, fd_monitor=closed}}.
 
 
+format_status(normal, [_PDict, #st{} = St]) ->
+    [{data, [{"State", St}]}];
+
+format_status(terminate, [_PDict, #st{} = St]) ->
+    [{data, [{"State", reduce_record(St)}]}].
+
+
+reduce_record(#st{} = St) ->
+    reduce_record(St, undefined).
+
+reduce_record(#st{} = St, _) ->
+    #st{
+        header = Header,
+        id_tree = IdTree,
+        seq_tree = SeqTree,
+        local_tree = LocalTree,
+        purge_tree = PurgeTree,
+        purge_seq_tree = PurgeSeqTree
+    } = St,
+    St#st{
+        header = reduce_record(Header, undefined),
+        id_tree = reduce_record(IdTree, undefined),
+        seq_tree = reduce_record(SeqTree, undefined),
+        local_tree = reduce_record(LocalTree, undefined),
+        purge_tree = reduce_record(PurgeTree, undefined),
+        purge_seq_tree = reduce_record(PurgeSeqTree, undefined)
+    };
+
+reduce_record(Header, _) when element(1, Header) =:= db_header ->
+    couch_bt_engine_header:reduce_record(Header);
+
+reduce_record(Tree, _) when element(1, Tree) =:= btree ->
+    {tree_size, couch_btree:size(Tree)};
```

Comment:
this redacts every part of the btree's and replaces it with just the tree size (which isn't part of the state). I'm not seeing anything sensitive in the btree attributes to warrant this.